lib/iris/analysis/trajectory.py

```diff
@@ -272,7 +279,8 @@ def interpolate(cube, sample_points, method=None):
         cache = {}
         column_indexes = _nearest_neighbour_indices_ndcoords(
             cube, sample_points, cache=cache)
-
+        if index_only:
+            return column_indexes
```

Comment:
@corinnebosley this is not ideal for two reasons:

* multiple return statements in a unit make for code that can be hard to use and debug as there are multiple exit points from the unit. This could mean you set a debug point that's never hit because it has been added after an earlier return statement. This situation is made exponentially worse with increased unit length - and especially in this case where it's hard to even _spot_ there's a return statement in the middle of the unit.
* Your two return statements return objects of different types (as @djkirkham has pointed out). This will cause difficulties where user code expects a cube to be returned but gets an array. On a related point, this second return value means the code no longer adheres to its docstring, which states this function "Extract[s] a sub-cube", which is no longer always true.